__tests__/app.test.js

```diff
@@ -82,4 +83,42 @@ describe('Testing app', () => {
                 })
         });
     });
+    describe('GET /api/articles/:article_id', () => {
+        test('status: 200, responds with a single article', () => {
+          return request(app)
+            .get(`/api/articles/1`)
+            .expect(200)
+            .then(({ body: { article } }) => {
+                article.forEach((oneArticle) => {
+                  expect(oneArticle).toEqual(
+                    expect.objectContaining({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    body: expect.any(String),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number)
+                     })
+                  );
+                });
+            });
+        });
+        test('status: 400, when an invalid article id is entered', () => {
+            return request(app)
+            .get(`/api/articles/barry`)
+            .expect(400)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Wrong Input!")
+            })
+        });
+        test('status: 404, when a valid type for article id is entered, however it currently does not exist', () => {
+            return request(app)
+            .get(`/api/articles/999999`)
+            .expect(404)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Woah! We're not that big yet!")

```

Comment:
fun message but not helpful if article one has been deleted and then I try and get it. Just tell them that that article_id doesn't exist

Reply:
Altered to ID Does Not Exist